src/maxdiffusion/generate_wan.py

```diff
@@ -132,17 +117,9 @@ def inference_generate_video(config, pipeline, filename_prefix=""):
 def run(config, pipeline=None, filename_prefix=""):
   print("seed: ", config.seed)
   model_key = config.model_name
-
-  checkpointer_lib = get_checkpointer(model_key)
-  WanCheckpointer = checkpointer_lib.WanCheckpointer
-
-  checkpoint_loader = WanCheckpointer(config, "WAN_CHECKPOINT")
+  checkpoint_loader = WanCheckpointer(model_key=model_key, config=config)
```

Comment:
before loading pipeline in line 122, should check if pipeline is None. Else when this is used in the training pipeline, it will re-load the weights again even though the training pipeline already loaded them.